sql/src/main/java/org/apache/druid/sql/calcite/schema/SchemaConfig.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.schema;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class SchemaConfig
+{
+  @JsonProperty
+  private final int tasksLimit;
```

Comment:
`OverlordResource.getTasks()` which is the API used by the `tasks` table accepts the `max` param to limit the number of tasks returned. How about adding this config in `TaskStorageConfig` which would be used as the default when the `max` param is missing? This will change the behavior of the `getTasks()` API, but I think that's OK.

Reply:
TaskStorageConfig is in `indexing` module, and I will have to add it as dependency for `sql` module because of that change.

Reply:
Should I move TaskStorageConfig to `core` instead?

Reply:
Perhaps I should have been clearer. [`TaskStorageConfig` is already being used to provide the default value for `durationBeforeNow` for `getRecentlyCreatedAlreadyFinishedTaskInfo()`](https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/overlord/MetadataTaskStorage.java#L232) which is the method used to compute the task statuses to return in `OverlordResource.getTasks()`. You don't have to add a new dependency, but get the default value from `TaskStorageConfig` when `maxTaskStatuses` is null at https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/overlord/MetadataTaskStorage.java#L233.

Reply:
So you propose to limit the result in OverlordResource instead, right?

Reply:
Yes, that's my suggestion.